app/controllers/employment/CheckEmploymentDetailsController.scala

```diff
@@ -91,8 +91,11 @@ class CheckEmploymentDetailsController @Inject()(implicit val cc: MessagesContro
             if (!cya.isPriorSubmission && !cya.employment.employmentDetails.isFinished) {
               Future.successful(RedirectService.employmentDetailsRedirect(cya.employment, taxYear, employmentId, cya.isPriorSubmission))
             } else {
-              Future.successful(performAuditAndRenderView(cya.employment.toEmploymentDetailsView(
-                employmentId, !cya.employment.employmentDetails.currentDataIsHmrcHeld), taxYear, isInYear))
+              prior match {
+                case Some(employment) => Future.successful(performAuditAndRenderView(cya.employment.toEmploymentDetailsView(
+                  employmentId, !cya.employment.employmentDetails.currentDataIsHmrcHeld), taxYear, isInYear, employment))
+                case None => Future.successful(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
```

Comment:
I think we need to check the outcome of this function, as we are now seeing failures in the journey tests relating to the match of prior data here.

Reply:
yeah prior data is not mandatory here so shouldn't be redirecting if no prior data